src/main/java/uk/ac/ebi/eva/pipeline/io/VepProcess.java

```diff
@@ -0,0 +1,341 @@
+                checkOutputWritingStatus();
+            } finally {
+                process = null;
+                logger.trace("VEP process finished");
+            }
+        }
+    }
+
+    private void flushToPerlStdin() {
+        try {
+            processStandardInput.flush();
+            processStandardInput.close();
+        } catch (IOException e) {
+            logger.error("Could not close stream for VEP's stdin", e);
+        }
+    }
+
+    private void waitUntilProcessEnds(Long timeoutInSeconds) {
+        boolean finished;
+        try {
+            boolean processWroteDuringWait;
+            do {
+                long beforeWaiting = System.currentTimeMillis();
+                finished = process.waitFor(timeoutInSeconds, TimeUnit.SECONDS);
+                processWroteDuringWait = beforeWaiting < outputIdleSince.get();
+                if (processWroteDuringWait && !finished) {
+                    logger.debug("Extending the timeout, as the process wrote more lines (it's still active)");
+                }
+            } while (processWroteDuringWait && !finished);
+        } catch (InterruptedException e) {
+            throw new ItemStreamException(e);
+        }
+
+        if (!finished) {
+            String timeoutReachedMessage = "VEP has been idle for more than the timeout (" + timeoutInSeconds
+                    + " seconds). Killed the process.";
+            logger.error(timeoutReachedMessage);
+            process.destroy();
+            throw new ItemStreamException(timeoutReachedMessage);
+        }
+    }
+
+    private void checkExitStatus() {
+        int exitValue = process.exitValue();
+        if (exitValue != 0) {
+            String timestamp = Long.toString(System.currentTimeMillis());
+            String backupVepOutput = annotationParameters.getVepOutput().replaceFirst("tsv\\.gz$",
+                    timestamp + ".tsv.gz");
+
+            String errorLog = backupVepOutput + ".errors.txt";
+            try {
+                connectStreams(process.getErrorStream(), new FileOutputStream(errorLog));
+            } catch (IOException e) {
+                throw new ItemStreamException("VEP exited with code " + exitValue
+                        + " but the file to dump the errors could not be created: " + errorLog,
+                        e);
+            }
+            boolean renamed = new File(annotationParameters.getVepOutput()).renameTo(new File(backupVepOutput));
+            if (renamed) {
+                logger.info("Renamed VEP output to " + backupVepOutput + " as backup because VEP failed");
+            }
+            throw new ItemStreamException("Error while running VEP (exit status " + exitValue + "). See "
+                    + errorLog + " for the errors description from VEP.");
+        }
+    }
+
+    private void checkOutputWritingStatus() {
+        try {
+            outputCapturer.join(timeoutInSeconds * CONVERT_SECONDS_TO_MILLISECONDS);
```

Comment:
Maybe a bit extreme? `timeoutInSeconds * 1000` would be clearly in milliseconds.